### PR TITLE
Login fixes

### PR DESCRIPTION
**Pull request by jfonte-accesso**
This is a package of fixes for login and for preflight checks sent by the Angular 2 app.
This does not address the use of sessions.

### CODE REVIEW

File: src/main/java/com/accesso/challengeladder/model/LoginForm.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package com.accesso.challengeladder.model;
```
Comment (olopez-accesso) [inline src/main/java/com/accesso/challengeladder/model/LoginForm.java]: this class should be moved to the package 'com.accesso.challengeladder.requests' where the PostMatchesRequest is or the other way around, so we've all the form/request classes in the same package.

File: src/main/java/com/accesso/challengeladder/services/UserService.java [hunk 2]
```diff
@@ -88,34 +88,15 @@ public boolean updateUser(User updatedUser)
...
-        return user;
-    }
-
     public User getUser(String userId) throws SQLException
     {
         User user = null;
         if (userId != null)
         {
             user = userDao.queryForId(userId);
+            getMatchRecord(user);
```
Comment (olopez-accesso) [inline src/main/java/com/accesso/challengeladder/services/UserService.java]: I will personally rename the getMatchRecord method to reflect that is updating the User object (i.e.: updateUserWinLosses) or make it returns the object so it's implicit what it does.

Comment (jfonte-accesso) [inline src/main/java/com/accesso/challengeladder/services/UserService.java]: OK thank you.

File: src/main/java/com/accesso/challengeladder/controller/LoginController.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+import java.io.IOException;
+import java.sql.SQLException;
+
+public class LoginController
+{
+    public LoginController() throws SQLException, IOException
+    {
+        UserService userService = new UserService();
+        post("/login", (req, res) -> {
```
Comment (olopez-accesso) [inline src/main/java/com/accesso/challengeladder/controller/LoginController.java]: I wouldn´t worry about changing this one at this point. Endpoints should be always resources, once we have HTTP Basic Authentication in place we could change this to "POST /token", every time we are authenticating a user we´ll be hitting this endpoint to generate a new token.

Comment (jfonte-accesso) [inline src/main/java/com/accesso/challengeladder/controller/LoginController.java]: What do you want me to change here? should i roll back the changes to create the login endpoint?

Comment (olopez-accesso) [inline src/main/java/com/accesso/challengeladder/controller/LoginController.java]: No, I would keep it as it's, but we might change it in the future if we decide to go the Basic auth +token route

**Pull request merged**